src/yaml.cpp

```diff
@@ -105,6 +105,34 @@ estd::result<YAML::Node, estd::error> readYamlFile(std::string const & path) {
 	return YAML::Load(buffer.str());
 }
 
+estd::result<void, estd::error> mergeYamlNodes(YAML::Node map_a, YAML::Node map_b) {
+	// Check if the arguments are maps.
+	if (!map_a.IsMap() || !map_b.IsMap()) {
+		return estd::error{"invalid argument type"};
+	}
+
+	YAML::const_iterator iterator = map_b.begin();
+	while (iterator != map_b.end()) {
+		std::string key = iterator->first.as<std::string>();
+		YAML::Node value = iterator->second;
+		if (iterator->second.IsMap()) {
+			if (map_a[key]) {
+				auto merged = mergeYamlNodes(map_a[key], value);
+				if (!merged) {
+					return estd::error{"failed to overwrite map: ", merged.error().format()};
```

Comment:
Shouldn't format the original error put pass it up, with some added description about which node cause the problem. Quite possibly `YamlError` is better suited for this, which can properly record the full path to a node.